File: sorting/bubble sort.cpp
/*
bubble sort in c++
*/

#include <bits/stdc++.h>
using namespace std;

void bubbleSort(int A[], unsigned int n) {
    for(int i = 0; i < n-1; i++) {
        for(int j = 0; j < n-i-1; j++) {
            if(A[j] > A[j+1]) {
                swap(A[j], A[j+1]);
            }
        }
    }
}

int main() {
    // randomize the array
    unsigned int size;
    int A[50000];
    srand(1);
    for (int i = 0; i < 50000; i++) {
        A[i] = rand() % (int)pow(2, 16) + 1;
    }

    // sort it
    bubbleSort(A, 50000);

    // print it
    for (auto i: A) {
        cout << i << ' ';
    }
    return 0;
}

File: sorting/bubble sort.py
def bubble_sort(h):
    n = len(h)
    for i in range(n-1):
        for j in range(n-i-1):
            if h[j] > h[j + 1]:
                h[j], h[j + 1] = h[j + 1], h[j]


def main():
    # randomize list
    import random
    random.seed("yeah ok bro")
    a = list(range(50000))
    random.shuffle(a)

    # sort it
    bubble_sort(a)

    # print it
    print(a)


if __name__ == "__main__":
    main()
